File: client/src/main/java/com/jraska/github/client/logging/FirebaseCrashReporter.java
package com.jraska.github.client.logging;

public final class FirebaseCrashReporter implements CrashReporter {
  private final FirebaseCrashProxy crashProxy;

  public FirebaseCrashReporter() {
    this(new FirebaseCrashProxy());
  }

  FirebaseCrashReporter(FirebaseCrashProxy crashProxy) {
    this.crashProxy = crashProxy;
  }

  @Override public CrashReporter report(Throwable error, String... messages) {
    for (String message : messages) {
      crashProxy.log(message);
    }

    crashProxy.report(error);
    return this;
  }
}

File: client/src/main/java/com/jraska/github/client/logging/FirebaseCrashReporter.kt
package com.jraska.github.client.logging

class FirebaseCrashReporter internal constructor(private val crashProxy: FirebaseCrashProxy) : CrashReporter {
  constructor() : this(FirebaseCrashProxy()) {}

  override fun report(error: Throwable, vararg messages: String): CrashReporter {
    for (message in messages) {
      crashProxy.log(message)
    }

    crashProxy.report(error)
    return this
  }
}
